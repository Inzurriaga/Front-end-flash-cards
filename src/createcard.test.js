import React from 'react';
import Create from './CreateCard.js';
import { shallow } from 'enzyme';

const CreateCard = jest.fn();
const cards = [{"catagory":"html","question":"What are the three main tags that all HTMl structure needs?","answer":"<html>, <head>, and <body>","falseAnswer":["<header>, <section>, and <footer>","<div>, <h1>, and <p>","<section>, <div>, and <span>"],"color":"#f1652a"},{"catagory":"html","question":"How do you insert an image?","answer":"<img>","falseAnswer":["<image>","<photo>","<jpeg>"],"color":"#f1652a"},{"catagory":"html","question":"What are the basic structure for a drop down menu?","answer":"<select> and <option>","falseAnswer":["<dropdown> and <option>","<input> and <option>","<input> and <dropdown>"],"color":"#f1652a"},{"catagory":"html","question":"How do you insert a video?","answer":"<video>","falseAnswer":["<vid>","<mp3>","<vhs>"],"color":"#f1652a"}];

describe('create', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Create cards={cards}/>);
  });

  it('should start off with the correct default states ', () => {
    expect(wrapper.state()).toEqual({
        personalDeck: []
    });
  });
  it('should display create card component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke the createCard method when clicked', () => {
    wrapper.find('.create-card-button').simulate('click');
    expect(CreateCard).toBeCalled();
  });
});