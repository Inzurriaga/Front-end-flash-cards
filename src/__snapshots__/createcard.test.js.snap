// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create should display create card component 1`] = `
<section>
  <h2
    className="create-card-title"
  >
    Create a Card
  </h2>
  <div
    className="create-card-container"
  >
    <div
      className="create-card"
    >
      <h4>
        Create Question/Defintion
      </h4>
      <textarea
        className="question-text"
      />
    </div>
    <div
      className="create-card"
    >
      <h4>
        Create Answer/Word
      </h4>
      <textarea
        className="answer-text"
      />
    </div>
  </div>
  <h3
    className="pick-label"
  >
    Pick a color
  </h3>
  <div
    className="pick-container"
  >
    <div>
      <input
        className="radio isHidden"
        id="radio0"
        name="create"
        type="radio"
        value="#f1652a"
      />
      <label
        className="label"
        for="radio0"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio1"
        name="create"
        type="radio"
        value="#2fa4d5"
      />
      <label
        className="label"
        for="radio1"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio2"
        name="create"
        type="radio"
        value="#f7d33c"
      />
      <label
        className="label"
        for="radio2"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio3"
        name="create"
        type="radio"
        value="#8dbf28"
      />
      <label
        className="label"
        for="radio3"
      >
         
      </label>
    </div>
  </div>
  <button
    className="create-card-button"
    onClick={[Function]}
  >
    Create Card
  </button>
</section>
`;

exports[`teaList should display buttons, instructions, and svgs 1`] = `
<section>
  <h2
    className="create-card-title"
  >
    Create a Card
  </h2>
  <div
    className="create-card-container"
  >
    <div
      className="create-card"
    >
      <h4>
        Create Question/Defintion
      </h4>
      <textarea
        className="question-text"
      />
    </div>
    <div
      className="create-card"
    >
      <h4>
        Create Answer/Word
      </h4>
      <textarea
        className="answer-text"
      />
    </div>
  </div>
  <h3
    className="pick-label"
  >
    Pick a color
  </h3>
  <div
    className="pick-container"
  >
    <div>
      <input
        className="radio isHidden"
        id="radio0"
        name="create"
        type="radio"
        value="#f1652a"
      />
      <label
        className="label"
        for="radio0"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio1"
        name="create"
        type="radio"
        value="#2fa4d5"
      />
      <label
        className="label"
        for="radio1"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio2"
        name="create"
        type="radio"
        value="#f7d33c"
      />
      <label
        className="label"
        for="radio2"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio3"
        name="create"
        type="radio"
        value="#8dbf28"
      />
      <label
        className="label"
        for="radio3"
      >
         
      </label>
    </div>
  </div>
  <button
    className="create-card-button"
    onClick={[Function]}
  >
    Create Card
  </button>
</section>
`;

exports[`teaList should display create card component 1`] = `
<section>
  <h2
    className="create-card-title"
  >
    Create a Card
  </h2>
  <div
    className="create-card-container"
  >
    <div
      className="create-card"
    >
      <h4>
        Create Question/Defintion
      </h4>
      <textarea
        className="question-text"
      />
    </div>
    <div
      className="create-card"
    >
      <h4>
        Create Answer/Word
      </h4>
      <textarea
        className="answer-text"
      />
    </div>
  </div>
  <h3
    className="pick-label"
  >
    Pick a color
  </h3>
  <div
    className="pick-container"
  >
    <div>
      <input
        className="radio isHidden"
        id="radio0"
        name="create"
        type="radio"
        value="#f1652a"
      />
      <label
        className="label"
        for="radio0"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio1"
        name="create"
        type="radio"
        value="#2fa4d5"
      />
      <label
        className="label"
        for="radio1"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio2"
        name="create"
        type="radio"
        value="#f7d33c"
      />
      <label
        className="label"
        for="radio2"
      >
         
      </label>
    </div>
    <div>
      <input
        className="radio isHidden"
        id="radio3"
        name="create"
        type="radio"
        value="#8dbf28"
      />
      <label
        className="label"
        for="radio3"
      >
         
      </label>
    </div>
  </div>
  <button
    className="create-card-button"
    onClick={[Function]}
  >
    Create Card
  </button>
</section>
`;
